### PR TITLE
[structure] Refactor pure neumann test case

### PR DESCRIPTION
**Pull request by maxfirmbach**
## Description and Context

In `4C` we have the ability to solve a pure neumann problem by projecting out it's rigid body modes and thus evade the singular nature of these types of setups.

1. Currently we have one test case for this in the old structure time integration. This PR refactors the case, removing all Dirichlet type boundary conditions that are not used anyway, to directly make clear that this is a pure neumann problem. In addition the test name is adapted.

2. This PR also adds the functionality to the new structure time integration and adds the same test case for it. The core functionality is just copied, so no major cleanup of the code etc. is provided.

### CODE REVIEW

File: src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp [hunk 1]
```diff
@@ -102,6 +104,98 @@ std::shared_ptr<Core::LinAlg::Solver> Solid::SOLVER::Factory::build_structure_li
       Teuchos::getIntegralValue<Core::IO::Verbositylevel>(
           Global::Problem::instance()->io_params(), "VERBOSITY"));
 
+
+  // setup Krylov projection if necessary
+  std::shared_ptr<Core::LinAlg::KrylovProjector> projector = nullptr;
+  {
+    // sysmat might be singular, e.g. when solid is not fully supported
+    // in this case, we need a basis vector for the nullspace/kernel
```
Comment (mayrmt) [inline src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp]:
```suggestion
    // Matrix might be singular, e.g. when solid is not fully supported in a static simulation.
    // In this case, we need a basis vector for the nullspace/kernel.
```

**Review by mayrmt — APPROVED** [review]

File: src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp (hunk 1 above)
Comment (maxfirmbach) [reply src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp]: Changed the documentation.

**Review by mayrmt — DISMISSED** [review]
LGTM!

File: src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp [hunk 2]
```diff
@@ -102,6 +104,98 @@ std::shared_ptr<Core::LinAlg::Solver> Solid::SOLVER::Factory::build_structure_li
...
+      }
+
+      const std::string* weighttype = &kspcond->parameters().get<std::string>("WEIGHTVECDEF");
+
+      projector = std::make_shared<Core::LinAlg::KrylovProjector>(
+          activemodeids, weighttype, actdis.dof_row_map());
+      weighttype = projector->weight_type();
+
+      if (*weighttype == "integration")
+        FOUR_C_THROW("Krylov projection an currently only be done pointwise.");
```
Comment (dharinib98) [inline src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp]: Typo in the error message. 
```suggestion
        FOUR_C_THROW("Krylov projection can currently only be done pointwise.");
```

**Comment by dharinib98** [comment]
@maxfirmbach Just a minor typo, otherwise good

File: src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp (hunk 2 above)
Comment (maxfirmbach) [reply src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp]: Fixed the spelling :+1:.
